configmap/hash-gen/main.go

```diff
@@ -68,19 +68,24 @@ func process(data []byte) ([]byte, error) {
 		return nil, nil
 	}
 
-	labels := traverse(content, "metadata", "labels")
-	if labels == nil {
-		// TODO(markusthoemmes): Potentially handle missing metadata and labels?
-		return nil, errors.New("'metadata.labels' not found")
+	metadata := traverse(content, "metadata")
+	if metadata == nil {
+		return nil, errors.New("'metadata' not found")
+	}
+
+	annotations := traverse(metadata, "annotations")
+	if annotations == nil {
```

Comment:
Can you add a test for this, that adds the mapping node for annotations **and** the annotation itself?

Reply:
Yeah good idea, will do

Reply:
(actually the existing tests do that, but I’ll add one for the annotation node already existing)

Reply:
done, PTAL :)

Reply:
(actually, now I think about it, this was technically already covered - `add.yaml` covers the case where annotations don't already exist, `update.yaml` covers the case where they don't).

Reply:
You're right, I misread the diff. Feel free to drop the commit :joy:

Reply:
hehe ok :)